hooks/useVerifiedLocations.tsx

```diff
@@ -4,166 +4,113 @@ import {
 import { useState } from "react";
 
+type HandleLocationResponseOptions = {
+  extraParamsIcon?: string;
+};
+
+const handleLocationResponse = (
+  data: any,
+  setLocations: (newData: any) => void,
+  options: HandleLocationResponseOptions = {}
+) => {
+  if (!data) return;
+
+  const features = data.results.map((item: any) => {
+    let extra_params = {};
+    try {
+      extra_params = dJSON.parse(item.extra_parameters);
+
+      if (options?.extraParamsIcon) {
+        extra_params = {
+          ...extra_params,
+          icon: options.extraParamsIcon,
+        };
+      }
+    } catch (error) {
+      console.error(error);
+    }
+
+    return {
+      type: "Feature",
+      geometry: {
+        type: "Point",
+        coordinates: item.loc?.reverse(),
+      },
+      properties: extra_params,
+    };
+  });
+
+  setLocations(features);
+};
+
 export function useVerifiedLocations() {
   const [foodLocations, setFoodLocations] = useState<any[]>([]);
   const [ahbapLocations, setAhbapLocations] = useState<any[]>([]);
   const [hospitalLocations, setHospitalLocations] = useState<any[]>([]);
   const [teleteyitLocations, setTeleteyitLocations] = useState<any[]>([]);
   const [satelliteLocations, setSatelliteLocations] = useState<any[]>([]);
 
-  useSWR(FOOD_URL, dataFetcher, {
-    onSuccess: (data) => {
-      if (!data) return;
-
-      const features = data.results.map((item: any) => {
-        let extra_params = {};
-        try {
-          extra_params = dJSON.parse(item.extra_parameters);
-        } catch (error) {
-          console.error(error);
-        }
-
-        return {
-          type: "Feature",
-          geometry: {
-            type: "Point",
-            coordinates: item.loc?.reverse(),
-          },
-          properties: extra_params,
-        };
-      });
+  const [sahraKitchenLocations, setSahraKitchenLocations] = useState<any[]>([]);
+  const [pharmacyLocations, setPharmacyLocations] = useState<any[]>([]);
+  const [safeLocations, setSafeLocations] = useState<any[]>([]);
+  const [depremIhtiyacLocations, setDepremIhtiyacLocations] = useState<any[]>(
+    []
+  );
 
-      setFoodLocations(features);
-    },
+  useSWR(FOOD_URL, dataFetcher, {
+    onSuccess: (data) => handleLocationResponse(data, setFoodLocations),
   });
 
   useSWR(AHBAP_LOCATIONS_URL, dataFetcher, {
-    onSuccess: (data) => {
-      if (!data) return;
-
-      const features = data.results.map((item: any) => {
-        let extra_params = {};
-        try {
-          extra_params = dJSON.parse(item.extra_parameters);
-        } catch (error) {
-          console.error(error);
-        }
-
-        return {
-          type: "Feature",
-          geometry: {
-            type: "Point",
-            coordinates: item.loc?.reverse(),
-          },
-          properties: extra_params,
-        };
-      });
-
-      setAhbapLocations(features);
-    },
+    onSuccess: (data) => handleLocationResponse(data, setAhbapLocations),
```

Comment:
new data sources can be added as below.

Reply:
you can follow the instructions at following when you needed to add new data source:

1 -  add a new Channel and MapType(First letter must be uppercased)
2 - Pass the data from LeafletMap locations prop(it must be same as Channel and MapType, lowercased)
3 - Add a new item to LayerContent to see at Drawer
?? 3 - Every point has different properties, to match it between drawer and point data, use propertyMap at GenericCluster. u cannot see the drawer data if you do not map it over propertyMap.
4 - boom it works